tests/test_bindings.py

```diff
@@ -1,33 +1,47 @@
+'''TESTS BINDING'''
+
+import os
 import json
+import sys
 import yaml
-import vttv
 
-import sys
-import os
+import vttv
 
 # source dir is the directory a level above this file
 source_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
+# Read the YAML config file
 with open(f'{source_dir}/tests/test_bindings_conf.yaml', 'r') as stream:
-  try:
-    params = yaml.safe_load(stream)
-  except yaml.YAMLError as exc:
-    print(exc)
+    try:
+        params = yaml.safe_load(stream)
+    except yaml.YAMLError as exc:
+        print(exc)
+
+# Check main key is "visualization"
+if "visualization" not in params:
+    print("The YAML configuration file is not valid: missing required paramaters: visualization")
+    sys.exit(1)
 
 # make output_visualization_dir directory parameter absolute
-params["visualization"]["output_visualization_dir"] = os.path.abspath(params["visualization"]["output_visualization_dir"])
+if "output_visualization_dir" in params["visualization"]:
+    params["visualization"]["output_visualization_dir"] = os.path.abspath(
+      params["visualization"]["output_visualization_dir"]
+    )
 
 params_serialized = yaml.dump(params["visualization"])
 
-n_ranks = params["visualization"]["x_ranks"] * params["visualization"]["y_ranks"] * params["visualization"]["z_ranks"]
-rank_data = []
+n_ranks = params["visualization"]["x_ranks"]
+n_ranks *= params["visualization"]["y_ranks"]
+n_ranks *= params["visualization"]["z_ranks"]
 
+# Prepare rank data
+rank_data = []
 for rank in range(n_ranks):
-  with open(f'{source_dir}/data/lb_test_data/data.{rank}.json', 'r') as f:
-    data = json.load(f)
-
-  data_serialized = json.dumps(data)
+    with open(f'{source_dir}/data/lb_test_data/data.{rank}.json', 'r') as f:
```

Comment:
I think you might have a pylint warning here. (specify encoding to be utf-8 in the `open` arguments).